src/test/java/org/opensearch/knn/bwc/KNNBackwardsCompatibilityIT.java

```diff
@@ -0,0 +1,193 @@
+                    addKnnDoc(testIndexName, "1", testFieldName, new Float[]{6.0f, 6.0f});
+
+                    knnWarmup(Collections.singletonList(testIndexName));
+
+                    assertEquals(graphCountBefore + 1, getTotalGraphsInCache());
+
+                    break;
+                case MIXED:
+                    Assert.assertTrue(pluginNames.contains("opensearch-knn"));
+
+                    float[] queryVector = {10.0f, 10.0f};
+                    int k = 1;
+
+                    KNNQueryBuilder knnQueryBuilder = new KNNQueryBuilder(testFieldName, queryVector, k);
+                    Response resp = searchKNNIndex(testIndexName, knnQueryBuilder,k);
+                    List<KNNResult> results = parseSearchResponse(EntityUtils.toString(resp.getEntity()), testFieldName);
+
+                    assertEquals(results.size(), k);
+                    for(KNNResult result : results) {
+                        String round = System.getProperty("tests.rest.bwcsuite_round");
+                        if(round.equals("first")) {
+                            assertEquals("1", result.getDocId());
+                            int graphCountFirst = getTotalGraphsInCache();
+
+                            deleteKnnDoc(testIndexName, "1");
+                            addKnnDoc(testIndexName, "2", testFieldName, new Float[]{50.0f, 50.0f});
+                            addKnnDoc(testIndexName, "4", testFieldName, new Float[]{55.0f, 55.0f});
+
+                            knnWarmup(Collections.singletonList(testIndexName));
+                            assertEquals(graphCountFirst + 2, getTotalGraphsInCache());
+                        }
+                        else if(round.equals("second")) {
+                            assertEquals("2", result.getDocId());
+                            int graphCountSecond = getTotalGraphsInCache();
+
+                            deleteKnnDoc(testIndexName, "2");
+
+                            knnWarmup(Collections.singletonList(testIndexName));
+                            assertEquals(graphCountSecond, getTotalGraphsInCache());
+                        }
+                       else {
+                            assertEquals("4", result.getDocId());
+                            deleteKNNIndex(testIndexName);
+                        }
+                    }
+
+                    break;
+                case UPGRADED:
+                    Assert.assertTrue(pluginNames.contains("opensearch-knn"));
+
+                    int graphCountUpgraded = getTotalGraphsInCache();
+
+                    updateKnnDoc(testIndexName, "1", testFieldName, new Float[]{17.0f, 17.0f});
+
+                    addKnnDoc(testIndexName, "3", testFieldName, new Float[]{20.0f, 20.0f});
+
+                    knnWarmup(Collections.singletonList(testIndexName));
+                    assertEquals(graphCountUpgraded+3, getTotalGraphsInCache());
```

Comment:
why 3?

Reply:
1. Adding knnDoc with id - 1
2. Updating knnDoc with id - 1
3. Adding knnDoc with id - 3

The updateKnnDoc operation is creating an extra graph and increasing the count. So, the total graph count is 3.